src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> tail;
+    private Node<T> head;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node newNode = new Node(null, value, null);
+            head = newNode;
+            tail = newNode;
+        } else {
+            Node newNode = new Node(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        Node<T> currentNode;
+        if (index == size) {
+            add(value);
+            return;
+        }
+        currentNode = goToNodeByIndex(index);
+        if (index == 0) {
+            newNode = new Node(null, value, currentNode);
+            size++;
+            head = newNode;
+            return;
+        }
+        newNode = new Node(currentNode.prev, value, currentNode);
+        size++;
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void addAll(List<T> myList) {
+        for (T itemFromList:myList) {
+            add(itemFromList);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> currentNode;
+        currentNode = goToNodeByIndex(index);
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode;
+        currentNode = goToNodeByIndex(index);
+        Node<T> oldNode = new Node(null, currentNode.value,null);
+        currentNode.value = value;
+        return oldNode.value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexOutOfSize(index);
+        Node<T> oldNode;
+        Node<T> currentNode;
+        if (index == 0 && size > 1) {
+            oldNode = new Node(null, head.value,null);
+            unlink(head);
+        } else if (index == size - 1) {
+            oldNode = new Node(null, tail.value,null);
+            unlink(tail);
+        } else {
+            currentNode = goToNodeByIndex(index);
+            oldNode = new Node(null, currentNode.value, null);
+            unlink(currentNode);
+        }
```

Comment:
this whole logic should be moved to unlink method. You have to handle cases if the node to unlink is a head or tail